frontend/src/lib/skill_tree.ts

```diff
@@ -384,87 +384,72 @@ const tradeStatNames: { [key: number]: { [key: string]: string } } = {
-              disabled: c != conqueror,
-              value: {
-                min: r.seed,
-                max: r.seed
-              }
-            }))
-          )
-          .flat(),
-        disabled: false
+        stats: [{
+          type: "count",
+          value: { min: 1 },
+          filters: stats
+        }]
+      },
+      sort: {
+        price: 'asc'
       }
-    ];
-  } else {
-    const seedGroups = Math.ceil(result.length / maxQueries);
-    stats = Object.keys(tradeStatNames[jewel])
-      .filter((c) => (onlyConqueror ? c == conqueror : true))
-      .map((c) => {
-        const groups = [];
-
-        for (let i = 0; i < seedGroups; i++) {
-          groups.push({
-            type: 'count',
-            value: {
-              min: 1
-            },
-            filters: result.slice(i * maxQueries, (i + 1) * maxQueries).map((r) => ({
-              id: tradeStatNames[jewel][c],
-              disabled: false,
-              value: {
-                min: r.seed,
-                max: r.seed
-              }
-            })),
-            disabled: c != conqueror || i > 0
-          });
-        }
+    }
+  }
 
-        return groups;
+  let max_query_length = 200
+  let final_query: Array<Object> = []
+
+  // only one conqueror, used when clicking on the all items trade button
+  if (onlyConqueror) {
+    max_query_length = 160
+
+    for (const r of result) {
+      final_query.push({
+        id: tradeStatNames[jewel][conqueror],
+        value: {
+          min: r.seed,
+          max: r.seed
+        }
       })
-      .flat();
-  }
+    }
+  } else {
+    // all conquerors, used when clicking on the single item item trade button
+    for (const r of result) {
+      let cluster: Array<Object> = []
+
+      for (const conq of Object.keys(tradeStatNames[jewel])) {
+        cluster.push({
+          id: tradeStatNames[jewel][conq],
+          disabled: conq != conqueror,
+          value: {
+            min: r.seed,
+            max: r.seed
+          }
+        })
+      }
 
-  return {
-    query: {
-      status: {
-        option: 'online'
-      },
-      name: data.TimelessJewels[jewel],
-      type: 'Timeless Jewel',
-      stats
-    },
-    sort: {
-      price: 'asc'
+      final_query.push(cluster)
     }
-  };
-};
 
-export const openTrade = (jewel: number, conqueror: string, results: SearchWithSeed[]) => {
-  const url = new URL('https://www.pathofexile.com/trade/search/Affliction');
-  url.searchParams.set('q', JSON.stringify(constructQuery(jewel, conqueror, results)));
+    final_query = final_query.flat()
+  }
 
-  // POE Trade has a 32k limit
-  if (url.toString().length > 32000) {
-    console.warn('URL too long, generating only selected conqueror');
-    url.searchParams.set('q', JSON.stringify(constructQuery(jewel, conqueror, results, true)));
+  if (final_query.length > max_query_length) {
+    final_query = final_query.slice(0, max_query_length)
   }
 
+  return gen_final_query(final_query)
```

Comment:
Is there a reason why this is a function, if it only gets called once? Seems to just add confusion and misdirection.